internal/utils.go

```diff
@@ -101,14 +103,51 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]map[string
 		)
 	}
 
-	objectTemplates := make([]map[string]interface{}, 0, len(manifests))
+	policyTemplatesLength := 1
+	if policyConf.NonConsolidated {
+		policyTemplatesLength = len(manifests)
+	}
+	objectTemplates := make([]map[string]interface{}, 0, policyTemplatesLength)
+	policyTemplates := make([]map[string]map[string]interface{}, 0, policyTemplatesLength)
 	for _, manifest := range manifests {
 		objTemplate := map[string]interface{}{
 			"complianceType":   policyConf.ComplianceType,
 			"objectDefinition": manifest,
 		}
-		objectTemplates = append(objectTemplates, objTemplate)
+		if policyConf.NonConsolidated {
+			// casting each objTemplate with manifest to objectTemplates type
+			// build policyTemplate for each objectTemplates
+			policyTemplate := buildPolicyTemplatesItem(policyConf, &[]map[string]interface{}{objTemplate})
+			setNamespaceSelector(policyConf, policyTemplate)
+			policyTemplates = append(policyTemplates, *policyTemplate)
+		} else {
+			// put all objTemplate with manifest into single consolidated objectTemplates object
+			objectTemplates = append(objectTemplates, objTemplate)
+		}
 	}
+
+	//  just build one policyTemplate by using the above consolidated objectTemplates
+	if !policyConf.NonConsolidated {
+		policyTemplate := buildPolicyTemplatesItem(policyConf, &objectTemplates)
+		setNamespaceSelector(policyConf, policyTemplate)
+		policyTemplates = append(policyTemplates, *policyTemplate)
+	}
+
+	// check the enabled expanders and add additional policy templates
+	expandedPolicyTemplates := handleExpanders(manifests, policyConf)
+	policyTemplates = append(policyTemplates, expandedPolicyTemplates...)
+
+	return policyTemplates, nil
+}
+
+// void function to set namespaceSelector by reference.
+func setNamespaceSelector(policyConf *types.PolicyConfig, policyTemplate *map[string]map[string]interface{}) {
+	if policyConf.NamespaceSelector.Exclude != nil || policyConf.NamespaceSelector.Include != nil {
+		(*policyTemplate)["objectDefinition"]["spec"].(map[string]interface{})["namespaceSelector"] = policyConf.NamespaceSelector
+	}
+}
+
+func buildPolicyTemplatesItem(policyConf *types.PolicyConfig, objectTemplates *[]map[string]interface{}) *map[string]map[string]interface{} {
```

Comment:
Could you please add a comment describing the function in this style?
https://go.dev/blog/godoc

Reply:
Sure, will do all above